### PR TITLE
Sites Management Dashboard: Add design for empty states

### LINKED ISSUE #65173: Sites Management Dashboard: implement the designs and handling for empty states
Depends on: https://github.com/Automattic/wp-calypso/issues/65167

Let's handle the empty states in the sites list properly.

image
<br>
image

Done is:

- [ ] The UI matches the designs.
- [ ] The implementation relies on core components where possible or introduces the new ones in Calypso `/packages/components`.

### PR DESCRIPTION
**Pull request by vykes-mac**
#### Proposed Changes

* Add ability to show a specific empty state design for each site filter.

#### Testing Instructions

1. Check out the branch locally.
2. Set `sites = []` in `client/sites-dashboard/components/sites-container.tsx`
3. Navigate to /sites-dashboard and view the new Sites Dashboard.
4. Observe the different empty states
5. Test empty search state by entering a site that does not exist. It should show `No sites match your search.`

**Note:**  The illustration for create site state is not yet added and it's using the default illustration

* 
image

image

Related to [#65173](https://github.com/Automattic/wp-calypso/issues/65173)

### CODE REVIEW

File: client/sites-dashboard/components/sites-container.tsx
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-container.tsx]: Should we change `filter` to `status` for consistency with #65443?

**Review by danielbachhuber — APPROVED** [review]
Looks great! I merged `trunk` and resolved the merge conflict from #65443 for you.

I don't have a strong opinion on the React code structure, so it probably make sense to get a second review from someone who does.

One thing I noticed too (can be addressed in a separate PR): the "Create your first site" appears briefly when the page is loaded with a fresh cache. You can reproduce by clearing the `calypso_store` in IndexDB:

image

It seems like we'll need a loading state too.

**Comment by vykes-mac** [comment]
@danielbachhuber 

> I don't have a strong opinion on the React code structure, so it probably make sense to get a second review from someone who does.

Yeah I'll wait for a second opinion from @Automattic/build team members 

> One thing I noticed too (can be addressed in a separate PR): the "Create your first site" appears briefly when the page is loaded with a fresh cache. You can reproduce by clearing the calypso_store in IndexDB:

I did notice that also, as you've said it can be fixed at a later PR when we include loading state.

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: will update accordingly

File: client/sites-dashboard/components/sites-container.tsx
Comment (wojtekn) [inline client/sites-dashboard/components/sites-container.tsx:4]: Should we use another component and avoid importing one from Calypso?

**Review by wojtekn — COMMENTED** [review]
It works as expected. I don't have an opinion on React code structure as well, I've just added a comment on the component import.

File: client/sites-dashboard/components/sites-container.tsx
Comment (zaguiini) [inline client/sites-dashboard/components/sites-container.tsx]: The problem with this `height` property is that it might overflow on smaller, mobile screens. Setting `100%` won't work because the browser won't be able to calculate the remaining space, and `100vh` won't work as well because there are other elements taking part of the height, so we won't be able to make this content centered easily.

I suggest removing the property. The results aren't bad either and we will respect the user's viewport.

File: client/sites-dashboard/components/sites-container.tsx
Comment (zaguiini) [inline client/sites-dashboard/components/sites-container.tsx]: This surrounding element is useless. The `styled` function can accept a component too if it accepts the `className` property. So we can do something like:

```js
styled(EmptyContent)`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	height: 600px;
`
```

And remove that surrounding `EmptyContainer` 👌

File: client/sites-dashboard/components/sites-container.tsx
Comment (zaguiini) [inline client/sites-dashboard/components/sites-container.tsx]: Though it's fine to use a `switch` statement in a render function in React, it's not a common pattern. People usually find `switch`es hard to read and they feel kinda off in this context. As we're already using an `if` above, let's try to have some consistency and keep using early-exit ifs (no `else`s too).

But don't worry, there's nothing wrong with it, it's just that it's not common to see and a bit hard to read.

File: client/sites-dashboard/components/sites-container.tsx
Comment (zaguiini) [inline client/sites-dashboard/components/sites-container.tsx]: Using nested selectors breaks modularity. Imagine you'd want to use this component but don't apply this line styling to a single instance of it. How would you do it? 🙂

So, my suggestion is to create a separate component (`styled.p`), add the `max-width` property there and use this component inside the `line` prop in `<EmptyContent />` instead.

**Review by zaguiini — COMMENTED** [review]
Works fine! I left a few comments.

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: yeah I was wondering how to get it centered otherwise. we probably would have to use top margin instead and transform by 50% of viewport height?

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: Thanks for this piece of insight 👍

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: yeah switches are not my favourite conditionals. Lets see if the ifs will read better

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: Thanks for insight again, makes a lot of sense, will update accordingly

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx:4]: I'm not sure this is an issue 😕 ? I see other places we import from calypso c`lient/sites-dashboard/components/sites-table-row.tsx`

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [inline client/sites-dashboard/components/sites-container.tsx:19]: @zaguiini What i did for now to kind of center states that doesn't have an illustration is to give the title a 50% margin from the top.

File: client/sites-dashboard/components/sites-container.tsx
Comment (wojtekn) [reply client/sites-dashboard/components/sites-container.tsx:4]: There was a discussion that ended up with the conclusion that we shouldn't have dependencies on Calypso. Trying to clarify it on p1657261784919289-slack-C0347E545HR

Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx:4]: Thank you for the discussion thread. I'm wondering though since these components exist it feels kinda weird recreating them. Maybe we should move them to `packages/` but that might not be in the scope of this work in terms of impact it might have if we move them.

Comment (danielbachhuber) [reply client/sites-dashboard/components/sites-container.tsx:4]: To avoid blocking this pull request, I think we might want to move this to a follow-up issue.

File: client/sites-dashboard/components/sites-table-filter-tabs.tsx
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-table-filter-tabs.tsx]: Should this be `status`?

File: client/sites-dashboard/components/sites-container.tsx
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-container.tsx]: Looks like this `sites = [];` needs to be removed 😄

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx:4]: Yeah I agree

File: client/sites-dashboard/components/sites-container.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-container.tsx]: 😆 that's why we have code review 🤣

File: client/sites-dashboard/components/sites-table-filter-tabs.tsx
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-table-filter-tabs.tsx]: yep

**Comment by zaguiini** [comment]
@vykes-mac, could you please add the no results state (p1657784954075839/1657737169.703459-slack-C0347E545HR) here as well? Thanks!

We should show this state when there's a search term, but no results.

File: client/sites-dashboard/components/searchable-sites-table.tsx
Comment (vykes-mac) [inline client/sites-dashboard/components/searchable-sites-table.tsx]: I wanted to add this as a state in the `sites-container `component but I think handling only global empty states there make more sense. I would have to create some reference to `searchable-sites `state since it's a child component so I could determine if a search term is entered or not, then re-render the `sites-container` to display the no sites found state.

Comment (zaguiini) [reply client/sites-dashboard/components/searchable-sites-table.tsx]: That's ok. Remember to wrap the string with `__` so it gets translated 😉

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7439691

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7439691).

**Pull request merged**
